yarn-project/simulator/src/acvm/acvm.ts

```diff
@@ -56,7 +64,16 @@ export async function acvm(
         }
 
         const result = await oracleFunction.call(callback, ...args);
-        return typeof result === 'undefined' ? [] : [result];
+
+        if (typeof result === 'undefined') {
+          return [];
+        } else if (result instanceof Array && !result.every(item => typeof item === 'string')) {
+          // We are dealing with a nested array which means that we do not need it wrap it in another array as to have
+          // the nested array structure it is already "wrapped".
+          return result;
+        } else {
+          return [result] as ForeignCallOutput[];
+        }
```

Comment:
I didn't want to force us to always wrap the return value in array like we do on line 75 in Oracle.ts so I simply detect if the wrapping is needed or not. If I didn't do this there would be way more changes in Oracle.ts so I think it's an ok approach.